includes/widgets/common.php

```diff
@@ -981,6 +981,498 @@ protected function register_controls() {
 			]
 		);
 
+		$this->start_controls_tabs( '_tabs_positioning' );
+
+		foreach ( [ 'normal', 'hover' ] as $tab ) {
+			$hover_state = 'hover' === $tab ? ':hover' : '';
+
+			$this->start_controls_tab(
+				"_tab_positioning_{$tab}",
+				[
+					'label' => 'normal' === $tab ? __( 'Normal', 'elementor' ) : __( 'Hover', 'elementor' ),
+				]
+			);
+
+			$this->add_control(
+				"_{$tab}_transform_rotate_popover",
+				[
+					'label' => __( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::POPOVER_TOGGLE,
+					'prefix_class' => 'e-',
+					'return_value' => 'transform',
+				]
+			);
+
+			$this->start_popover();
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_rotateZ_effect",
+				[
+					'label' => __( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'deg' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-rotateZ: {{SIZE}}deg',
+					],
+					'condition' => [
+						"_{$tab}_transform_rotate_popover!" => '',
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			$this->add_control(
+				"_{$tab}_transform_rotate_3d",
+				[
+					'label' => __( '3D Rotate', 'elementor' ),
+					'type' => Controls_Manager::SWITCHER,
+					'label_on' => __( 'On', 'elementor' ),
+					'label_off' => __( 'Off', 'elementor' ),
+				]
+			);
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_rotateX_effect",
+				[
+					'label' => __( 'Rotate X', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'deg' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'condition' => [
+						"_{$tab}_transform_rotate_3d!" => '',
+						"_{$tab}_transform_rotate_popover!" => '',
+					],
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-rotateX: {{SIZE}}deg;',
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_rotateY_effect",
+				[
+					'label' => __( 'Rotate Y', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'deg' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'condition' => [
+						"_{$tab}_transform_rotate_3d!" => '',
+						"_{$tab}_transform_rotate_popover!" => '',
+					],
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-rotateY: {{SIZE}}deg;',
```

Comment:
The value that the variable get is between 0-100 deg, so if you set this control to 100 you get 100deg instead of 360deg.

Reply:
![image](https://user-images.githubusercontent.com/79147990/124436711-ee62c400-dd7e-11eb-9e8c-fc7509240eb4.png)

Reply:
Following our conversation, please check again.

Reply:
done